### DISCUSSION TITLE
Porting the library to OsX

### DISCUSSION THREAD

**Discussion by oto212** [alekmaul/pvsneslib#126] [Q&A] [answered]
Hi @alekmaul 
Props to you for this fantastic project

Any chance the library would be ported to OSX ?

**Comment by alekmaul** [comment, marked as answer]
Well, we need to work on compiler & linker, they are the only tools not compatible with OSX

**Reply by oto212** [reply]
Any chance that you guys work on it ?

**Reply by alekmaul** [reply]
We need to find someone with an OSX platform to work on it

**Comment by Dienben** [comment]
Hi,

First, I would like to thank you for this fantastic project! Jumping on the topic as a newbie, I would like to see how I can contribute.
I'm on Mac, and I just managed to generate rom (the hello world one :-) on my Mac (M1). 
I'm not super strong technically, but I will do my best to help! So if somebody wants to guide me, I will follow the rules!

So, bringing back to the Mac topic, my purpose was to be able to get the code, modify it, process the assets, run the compiler, linker, etc... I have dockerized the chain. So, from a user point of view:
1 - Install docker Desktop on your Mac
2 - run the docker build command (based on a specific docker file): I created one with all the dependencies (python3, gcc, etc...)
3 - launch the container with a local volume
3 - in the local volume, I've unzipped the Linux release of pvsneslib
4- My VScode is simply pointing in this local volume, so then I launch the make command (in the container) via the terminal

And it works! no fancy dependencies or commands...

I'm currently trying to compile Mesen2 (the make command doesn't work) for the debugger (any suggestion is welcome).

If you agree with the approach, how can I contribute? Any guidance? I also want to write small tutorials to help people start quickly.

Again, thank you so much for all your efforts for the community!

Ben
<img width="1727" alt="Screenshot 2023-02-17 at 12 21 53 PM" src="https://user-images.githubusercontent.com/4283549/219559362-baa57511-5c73-433a-8ac8-371f44e7f067.png">

**Comment by alekmaul** [comment]
Thanks a really great news, but (if I remember well) I got troubles when I wanted to compile the toolchain some time ago with github actions.
I will need to do a new try this week end. Thaks for the news !
